pkg/operator/operator/validations.go

```diff
@@ -744,3 +786,53 @@ func getValidationK8sResources() ([]kunstructured.Unstructured, *kresource.Quant
 
 	return virtualServices, maxMem, err
 }
+
+func validateOverridenImages(images, keys []string) error {
+	images = slices.RemoveEmpties(images)
+	keys = slices.RemoveEmpties(keys)
+
+	ECRImages := make([]bool, len(images))
+	for i, image := range images {
+		if regex.IsValidECR(image) {
+			ECRImages[i] = true
+		}
+	}
+
+	ECRAuthConfig := aws.ECRAuthConfig{}
+	dockerECRAuthConfig := dockertypes.AuthConfig{}
+	if slices.HasTrue(ECRImages) {
+		tokenOutput, err := aws.GetECRAuthToken()
+		if err != nil {
+			return err
+		}
+		ECRAuthConfig, err = aws.ExtractECRAuthConfigFromTokenOutput(tokenOutput)
+		if err != nil {
+			return err
+		}
+
+		dockerECRAuthConfig.Username = ECRAuthConfig.Username
+		dockerECRAuthConfig.Password = ECRAuthConfig.AccessToken
+		dockerECRAuthConfig.ServerAddress = ECRAuthConfig.ProxyEndpoint
+	}
+
+	ECRAuth, err := dockerlib.EncodeAuthConfig(dockerECRAuthConfig)
+	if err != nil {
+		return err
+	}
+	noAuth, _ := dockerlib.EncodeAuthConfig(dockertypes.AuthConfig{})
```

Comment:
This could be moved into a global variable in lib/docker, e.g. dockerlib.NoAuth, do you think that makes sense?

Reply:
Yeah, I could see it being used multiple times.